### ISSUE TITLE
[Assistive tech and device support] Keyboard navigation order isn't meaningful or sequential. (11.08.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#69121]
#### VFS team name
Benefit Tools

#### VFS product name
Claim Status Tool

#### VFS feature name
Improved Evidence Request Initiative

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Keyboard navigation order isn't meaningful or sequential.

### Issue Details
I'm noticing a keyboard navigation issue that definitely seems unintentional, but is going to cause significant issues for keyboard-only users. When [/track-claims/your-claims](https://staging.va.gov/track-claims/your-claims) loads, keyboard focus appears to go straight to the "Your claims, decision reviews, or appeals" section while the visual viewport stays at the top of the page. When you press [tab], focus moves to the first focusable element after the additional info component (either the first "View details" link or the "Change your address" link if the claim cards haven't finished loading yet.

The unexpected jump can be disorienting (wait, how did I end up way down here?) and adds friction if you're trying to reach the mobile app download links. For screen reader users, they may not realize the mobile app download links exist at all.

### Link, screenshot or steps to recreate
1. Navigate to /track-claims/your-claims.
2. After page load, press [tab].
3. Observe that keyboard focus jumps to a link halfway down the page.

### VA.gov Experience Standard
[Category Number 11, Issue Number 08](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 2.4.3_A

### Platform Recommendation
Initial focus should stay at the top of the page --- or, if otherwise, stay consistent with what's visible in the viewport.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by davidakennedy** [comment]
I did some testing on this today. Here are some notes:

I can definitely reproduce the issue, as shown in the GIF:

![Focus on claim status tool moves to loading indicator, then to body after the page loads.](https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/b840c0e7-9097-4562-a5a8-a2ad693171de)

What happens in the GIF:

You can see where focus goes because of some debugging I have in the browser console.

1. As the page is loading, focus shifts to the loading indicator. This seems possible, thanks to a [`setFocus` property in the component](https://design.va.gov/components/loading-indicator#code-usage).
2. Then focus shifts back to the body.
3. When you tab, it goes to the first link in the first card.

My working theory is we're setting focus on the loading indicator and that's throwing things off somehow. @jerekshoe @samcoforma Are we doing that at all?

**Comment by jerekshoe** [comment]
IIRC the loading indicator is part of a platform component that sets focus on itself when it is rendered. I’m not sure what we could do to fix that without updating the platform code…

**Comment by davidakennedy** [comment]
@jerekshoe looked into this more yesterday, and we chatted about next steps today. Here are some notes and next steps:

**Notes:**

- The bug isn't really a bug, but being caused by focus shifting to a loading indicator that appears before the claims list loads in. It captures focus, per it's property, but it isn't sending focus elsewhere after the list loads in.
- We could send focus elsewhere or make it so the loading indicator doesn't capture focus.
- Currently, some loading indicators in our team's apps capture focus and some don't. Unknown: How do other apps across VA.gov handle this?
- From an accessibility perspective, it's good for a low vision or blind Veteran to know if something is loading because they may not be able to see it. However, with multiple loading indicators potentially firing at the same time or sequentially, it could be confusing.

**Next steps:**

- Jerek will remove the focus property on the loading indicator, which should prevent focus from skipping the top of the page, as noted in this ticket.
- DK will create a ticket to do a spike around loading indicators across Benefits apps and VA.gov to help firm up guidance in the design system and consistency on VA.gov.

**Comment by davidakennedy** [comment]
@jerekshoe Retested this today and it works. After the loading indicators finish, the first tab stop is the list of "on this page" links. See GIF with me testing it in Chrome. Also took a look in Firefox and Safari.

![Focus on claim status tool moves to body element after page loads, then focus is tabbed to list of "on this page" links.](https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/302d7fbc-e017-456f-a2f9-4a29e899f853)

**Comment by HeatherWidmont** [comment]
@davidakennedy will reach out to Brian to test & ensure this works

**Comment by davidakennedy** [comment]
Brian [retested this and said it looks good](https://dsva.slack.com/archives/C04KHCT3ZMY/p1701195774540009?thread_ts=1701193401.075019&cid=C04KHCT3ZMY).
